src/Progress/Store.js

```diff
@@ -1,43 +1,81 @@
-import _ from 'lodash/fp'
 import { combineReducers } from 'redux'
 import BaseStore from '../Core/BaseStore'
 
 export const identifier = 'progress'
+
 const baseStore = BaseStore(identifier)
 
-const isVisible = d => d.code < 200 || d.code >= 400
-const getTotal = _.sumBy( () => 1 )
-const getDone = _.sumBy( e => e.erfuellt )
-const dashboardData = _.flow([
-    _.over([ getTotal, getDone ]),
-    ([ total, done ]) => ({ total, done })
-])
+function moduleIsVisible(module) { return module.code < 200 || module.code >= 400; }
+
+function groupBy(data, key) {
+
+    return data.reduce((storage, item) => {
+
+        var group = item[key] - 1;
+        storage[group] = storage[group] || [];
+        storage[group].push(item);
+        return storage;
+
+    }, []);
+}
+
+function getStudienfortschrittDisplayData(studienleistungen) {
+
+    studienleistungen = studienleistungen.meilensteine;
+    studienleistungen = groupBy(studienleistungen, 'fachsemester');
+
+    let studienfortschrittDisplayData = [];
+
+    studienleistungen.forEach(element => {
+
+        let object = {};
+        let zulassungsVoraussetzungFuerStudienleistung = element.find(d => d.code === element[0].fachsemester + 300);
+
+        object.label = element[0].fachsemester + '. Fachsemester';
+
+        object.prereq = !zulassungsVoraussetzungFuerStudienleistung || zulassungsVoraussetzungFuerStudienleistung.erfuellt;
+
+        object.completed = element.find(d => d.code === element[0].fachsemester + 200).erfuellt;
+
+        object.entries = element
+            .filter(moduleIsVisible)
+            .map(d => ({
+                ...d,
+                link: (d.format && `/exams/${d.format}s/${d.studiPruefungsId}`)
+            }));
+
+        studienfortschrittDisplayData.push(object);
+    })
+
+    return studienfortschrittDisplayData;
+}
+
+function getStudienfortschrittDashboardData(state) {
+
+    let studienleistungen = baseStore.getItems(state);
+
+    studienleistungen = studienleistungen.flatMap(d => d.entries);
+    studienleistungen = studienleistungen.filter(moduleIsVisible);
+
+    let done = studienleistungen.reduce((acc, elm) => elm.erfuellt ? ++acc : acc, 0);
```

Comment:
ok - in bezug auf die benennung "done" . doneCount würde ich trotzdem besser finden. aber da die variable sehr lokal ist, und in der nächsten zeilt mit getTotal zusammen steht, wird es etwas besser verständlich. dennoch: ein variablenname sollte möglichst im kontext der zeile, in der er auftaucht verständlich sein. es sollte keine weitere zeile für den kontext notwendig sein. 
die abkürzungen elm und acc finde ich einfach hässlich - sorry. aber ich vermute, dass du die kürze der reduce function schön findest - und das kann ich auch verstehen.